src/middlewared/middlewared/main.py

```diff
@@ -794,29 +795,25 @@ async def run(self, ws, request, conndata):
         return ws
 
     async def worker_kill(self, t_worker):
-        # If connection has been closed lets make sure shell is killed
-        if t_worker.shell_pid:
+        def worker_kill_impl():
+            # If connection has been closed lets make sure shell is killed
+            if t_worker.shell_pid:
```

Comment:
Actually, I can't remember details but does the inner method have access to the outer methods args without specifing `nonlocal t_worker` inside `worker_kill_impl`?

Reply:
@yocalebo yes, the inner method has access to all the variables from the outer scope, but if you assign to the outer scope variable `t_worker = t_worker + 1` it will become a local variable, so they added `nonlocal t_worker` to prevent this from happening